internal/controller/backup_controller.go

```diff
@@ -362,28 +365,36 @@ func (r *BackupReconciler) getCluster(
 	ctx context.Context,
 	backup *apiv1.Backup,
 	cluster *apiv1.Cluster,
-) *ctrl.Result {
+) (*ctrl.Result, error) {
+	contextLogger := log.FromContext(ctx)
+
 	clusterName := backup.Spec.Cluster.Name
 	err := r.Get(ctx, client.ObjectKey{
 		Namespace: backup.Namespace,
 		Name:      clusterName,
 	}, cluster)
 	if err == nil {
-		return nil
+		return nil, nil
 	}
 
 	if apierrs.IsNotFound(err) {
 		r.Recorder.Eventf(backup, "Warning", "FindingCluster",
 			"Unknown cluster %v, will retry in 30 seconds", clusterName)
-		return &ctrl.Result{RequeueAfter: 30 * time.Second}
+		return &ctrl.Result{RequeueAfter: 30 * time.Second}, nil
+	}
+
+	contextLogger.Error(err, "error getting cluster, proceeding to flag backup as failed.")
+
+	if flagErr := resourcestatus.FlagBackupAsFailed(ctx, r.Client, backup, nil,
+		fmt.Errorf("while getting cluster %s: %w", clusterName, err)); flagErr != nil {
+		contextLogger.Error(flagErr, "while flagging backup as failed, retrying...")
+		return nil, flagErr
 	}
 
-	_ = resourcestatus.FlagBackupAsFailed(ctx, r.Client, backup, nil,
-		fmt.Errorf("while getting cluster %s: %w", clusterName, err))
 	r.Recorder.Eventf(backup, "Warning", "FindingCluster",
 		"Error getting cluster %v, will not retry: %s", clusterName, err.Error())
 
```

Comment:
previously this event emitter was very misleading, given that the backup could have potentially not be flagged as failed the reconciler would have retried (eventually)